### PR TITLE
avoid agent.run for getSelectedPair

### PR DESCRIPTION
**Pull request by hugoArregui**
Procted the selected candidate pair by a RWMutex so we don't need to relay in the agent taskloop in order to use this method

### CODE REVIEW

File: agent.go [hunk 1]
```diff
@@ -629,15 +640,19 @@ func (a *Agent) validateSelectedPair() bool {
 // if no packet has been sent on that pair in the last keepaliveInterval
 // Note: the caller should hold the agent lock.
 func (a *Agent) checkKeepalive() {
-	if a.selectedPair == nil {
+	a.selectedPairMutex.RLock()
```
Comment (Sean-Der) [inline agent.go]: Use `getSelectedPair` instead?

File: agent.go [hunk 2]
```diff
@@ -610,14 +615,20 @@ func (a *Agent) taskLoop() {
 // validateSelectedPair checks if the selected pair is (still) valid
 // Note: the caller should hold the agent lock.
 func (a *Agent) validateSelectedPair() bool {
-	if a.selectedPair == nil {
+	a.selectedPairMutex.RLock()
```
Comment (Sean-Der) [inline agent.go]: use `getSelectedPair` instead?

**Comment by Sean-Der** [comment]
@hugoArregui we should probably get rid of the `run` construct all together. If you see any another performance wins/chance to do it I would say go for it!

**Comment by hugoArregui** [comment]
> @hugoArregui we should probably get rid of the `run` construct all together. If you see any another performance wins/chance to do it I would say go for it!

Agree!

**Pull request merged**